server/src/main/java/io/crate/execution/engine/aggregation/impl/IntervalSumAggregation.java

```diff
@@ -0,0 +1,149 @@
+            IntervalSumAggregation::new
+        );
+    }
+
+    private final Signature signature;
+    private final BoundSignature boundSignature;
+    private final int bytesSize;
+
+    @VisibleForTesting
+    private IntervalSumAggregation(Signature signature, BoundSignature boundSignature) {
+        this.signature = signature;
+        this.boundSignature = boundSignature;
+        this.bytesSize = DataTypes.INTERVAL.fixedSize();
+    }
+
+    @Nullable
+    @Override
+    public Period newState(RamAccounting ramAccounting,
+                      Version indexVersionCreated,
+                      Version minNodeInCluster,
+                      MemoryManager memoryManager) {
+        ramAccounting.addBytes(bytesSize);
+        return null;
+    }
+
+    @Override
+    public Period iterate(RamAccounting ramAccounting,
+                          MemoryManager memoryManager,
+                          Period state,
+                          Input<?>[] args) throws CircuitBreakingException {
+        return reduce(ramAccounting, state, DataTypes.INTERVAL.sanitizeValue(args[0].value()));
+    }
+
+    @Override
+    public Period reduce(RamAccounting ramAccounting, Period state1, Period state2) {
+        if (state1 == null) {
+            return state2;
+        }
+        if (state2 == null) {
+            return state1;
+        }
+        return periodAdd(state1, state2);
+    }
+
+    @Override
+    public Period terminatePartial(RamAccounting ramAccounting, Period state) {
+        return state;
+    }
+
+    @Override
+    public DataType<?> partialType() {
+        return boundSignature.returnType();
+    }
+
+    @Override
+    public Signature signature() {
+        return signature;
+    }
+
+    @Override
+    public BoundSignature boundSignature() {
+        return boundSignature;
+    }
+
+    @Override
+    public boolean isRemovableCumulative() {
+        return true;
+    }
+
+    @Override
+    public Period removeFromAggregatedState(RamAccounting ramAccounting,
+                                            Period previousAggState,
+                                            Input<?>[] stateToRemove) {
+        return periodSubtract(previousAggState, DataTypes.INTERVAL.sanitizeValue(stateToRemove[0].value()));
+    }
+
+    @Nullable
+    @Override
+    public DocValueAggregator<?> getDocValueAggregator(LuceneReferenceResolver referenceResolver,
+                                                       List<Reference> aggregationReferences,
+                                                       DocTableInfo table,
+                                                       List<Literal<?>> optionalParams) {
+        return null;
+    }
+
+    private static Period periodAdd(Period period1, Period period2) {
+        return period1.plus(period2);
+    }
+
+    public static Period periodSubtract(Period period1, Period period2) {
+        return period1.minus(period2);
+    }
```

Comment:
Why not inline these? It's easier to read without this indirection. The method names don't add any semantic meanings and it's not shorter either.